lesson14.md

```diff
@@ -0,0 +1,178 @@
+c) Dumb Object Mess
+d) Deliberate Opinionated Mode
+
+How do we get an element by Id?
+
+* document.giveMeElement("id")
+* document.elementById("id")
+* document.getElementById("id")
+* document.GetElementByIdPlease("id")
+
+--->
+
+---
+
+HTML:
+```html
+<div id="myDiv">Hello</div>
+<input id="myInput" type="number" value="42" />
+<button id="myButton">Click me!</button>
+```
+
+Select the elements from JavaScript:
+```js
+let myDivElement = document.getElementById("myDiv");s
+let myInputElement = document.getElementById("myInput");
+let myButtonElement = document.getElementById("myButton");
+```
+
+---
+
+```js
+let myDivElement = document.getElementById("myDiv");s
+console.log(typeof myDivElement); // what does it print?
+```
+
+```js
+"object"
+```
+<!-- .element: class="fragment" -->
+
+---
+
+So `getElementById()` returns an object. What can we do with objects?
+
+* Set or get properties
+<!-- .element: class="fragment" -->
+* Call methods
+<!-- .element: class="fragment" -->
+
+---
+
+### Setting or getting properties
+
+```js
+let myInputElement = document.getElementById("myInput");
+// get a property
+console.log(myInputElement.value); // the current value
+// set a property
+myInputElement.value = 43; // change the value from JavaScript
+```
+
+---
+
+### Example
+
+HTML:
+```html
+<input id="myInput" type="number"
+    min=0 max=10
+    placeholder="Enter a number"
+/>
+```
+
+Same from JavaScript:
+```js
+let myInputElement = document.getElementById("myInput");
+myInputElement.type = "number";
+myInputElement.min = 0;
+myInputElement.min = 10;
+myInputElement.placeholder = "Enter a number";
+```
+
+---
+
+### textContent
+
+The `textContent` property allows to get or set the text within a HTML Element:
+
+```html
+<div id="myDiv">This is the text content</div>
+```
+
+JavaScript:
+
+```js
+let myDivElement = document.getElementById("myDiv");
+// get the text content
+console.log(myDivElement.textContent);
+// set a new text content
+myDivElement.textContent = "This is the new text content";
+```
+
+---
+
+### Calling methods
+
+```js
+let myDivElement = document.getElementById("myDiv");
+
+// remove the element
+myDivElement.remove();
+```
+
+---
+
+### Assigning new methods
+
+```js
+let myButtonElement = document.getElementById("myButton");
+myButtonElement.onclick = function() {
+    console.log("my button was clicked!");
+}
+```
+
+---
+
+### Let's practice
+
+1. Create a few HTML Elements in your HTML file
+1. Create a button, and call a function in its `onclick` handler
+1. Change some of the attributes of your HTML Elements
+
+---
+
+### Let's write a shop!
+
+1. Create some HTML, with two (or more) products you would like to sell in your shop. Every product should have a small description, a price and a `<input type="number">` element for selecting the amount.
+1. Assign a function to each input element's `oninput` property.
+1. Use the `valueAsNumber` property of your input element to get the value as `number` (not string).
```

Comment:
I think that this step is not really clear

Reply:
ok, I've added a bit more description

Reply:
actually I was confused by `valueAsNumber`. So, just googled it, I had no idea it was an actual property 😅